### PR TITLE
Auth ui v2 - Add SSO login

### PR DESCRIPTION
**Pull request by tung2744**
ref #3593 

Screenshot 2024-01-22 at 6 27 05 PM

### CODE REVIEW

File: pkg/auth/handler/webapp/authflow_controller.go [hunk 1]
```diff
@@ -187,7 +187,7 @@ func (c *AuthflowController) HandleOAuthCallback(w http.ResponseWriter, r *http.
 		return
 	}
 
-	s, err := c.getWebSession(r)
+	s, err := c.Sessions.Get(state.WebSessionID)
```
Comment (tung2744) [inline pkg/auth/handler/webapp/authflow_controller.go:190]: This is a bug which I think also affects authflow v1. If we are not going to deploy v2 onto staging shortly, maybe we need to pick this fix to main.

Comment (louischan-oursky) [reply pkg/auth/handler/webapp/authflow_controller.go:190]: I could not reproduce this bug on latest of authflowv2

**Comment by louischan-oursky** [comment]
Other parts look good to me.

File: pkg/auth/handler/webapp/authflow_controller.go (hunk 1 above)
Comment (tung2744) [reply pkg/auth/handler/webapp/authflow_controller.go:190]: I tested again but cannot find how to reproduce the bug. Maybe I will drop this commit first.

**Comment by louischan-oursky** [comment]
Rebase.

**Comment by tung2744** [comment]
Rebased, thanks!

**Pull request merged**